src/lib/utils/calendar.cpp

```diff
@@ -34,6 +35,46 @@ std::tm do_gmtime(std::time_t time_val)
    return tm;
    }
 
+#if !defined(BOTAN_TARGET_OS_HAS_TIMEGM) && !defined(BOTAN_TARGET_OS_HAS_MKGMTIME)
+std::mutex ENV_TZ;
+
+std::time_t fallback_timegm(std::tm *tm)
+   {
+   std::time_t out;
+   std::string tz_backup;
+
+   ENV_TZ.lock();
+
+   // Store current value of env variable TZ
+   auto tz_env_pointer = getenv("TZ");
```

Comment:
Use explicit `const char*` here, it is more readable, and as `getenv` actually returns a mutable `char*` that would be the type given to `tz_env_pointer`. But this function can store it as a const and so should do so.

Prefix all calls to functions in the global namespace with `::` (eg `::getenv`, ::tzset`) so it is clear to the reader that we are calling system functions rather than internal library calls.